src/main/java/org/mskcc/cbio/portal/scripts/ImportCnaDiscreteLongData.java

```diff
@@ -22,52 +22,108 @@
+import org.mskcc.cbio.portal.model.GeneticProfile;
+import org.mskcc.cbio.portal.model.Sample;
+import org.mskcc.cbio.portal.util.CnaUtil;
+import org.mskcc.cbio.portal.util.ConsoleUtil;
+import org.mskcc.cbio.portal.util.FileUtil;
+import org.mskcc.cbio.portal.util.GeneticProfileUtil;
+import org.mskcc.cbio.portal.util.ProgressMonitor;
+import org.mskcc.cbio.portal.util.StableIdUtil;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
 import java.util.Optional;
-import java.util.*;
-import java.util.stream.*;
+import java.util.Set;
+import java.util.stream.Collectors;
 
-import static com.google.common.collect.Lists.*;
-import static java.lang.String.*;
+import static com.google.common.collect.Lists.newArrayList;
+import static java.lang.String.format;
 import static org.cbioportal.model.MolecularProfile.DataType.DISCRETE;
+import static org.cbioportal.model.MolecularProfile.ImportType.DISCRETE_LONG;
 
 public class ImportCnaDiscreteLongData {
 
     private final File cnaFile;
     private final int geneticProfileId;
-    private final GeneticAlterationImporter geneticAlterationGeneImporter;
+    private GeneticAlterationImporter geneticAlterationGeneImporter;
     private String genePanel;
     private final DaoGeneOptimized daoGene;
     private CnaUtil cnaUtil;
     private Set<CnaEvent.Event> existingCnaEvents = new HashSet<>();
     private int samplesSkipped = 0;
     private Set<String> namespaces;
 
+    private boolean updateMode;
+
+    private GeneticProfile geneticProfile;
+
     private final ArrayList<SampleIdGeneticProfileId> sampleIdGeneticProfileIds = new ArrayList<>();
+    private ArrayList<Integer> orderedSampleList;
 
     public ImportCnaDiscreteLongData(
-        File cnaFile,
-        int geneticProfileId,
-        String genePanel,
-        DaoGeneOptimized daoGene,
-        DaoGeneticAlteration daoGeneticAlteration,
-        Set<String> namespaces
+            File cnaFile,
+            int geneticProfileId,
+            String genePanel,
+            DaoGeneOptimized daoGene,
+            Set<String> namespaces,
+            boolean updateMode
     ) {
         this.namespaces = namespaces;
         this.cnaFile = cnaFile;
         this.geneticProfileId = geneticProfileId;
+        this.geneticProfile = DaoGeneticProfile.getGeneticProfileById(geneticProfileId);
+        if (!Set.of(DISCRETE.name(), DISCRETE_LONG.name()).contains(geneticProfile.getDatatype())) {
+            throw new IllegalStateException("Platform "
+                    + geneticProfileId
+                    + " has not supported datatype: "
+                    + geneticProfile.getDatatype());
+        }
         this.genePanel = genePanel;
         this.daoGene = daoGene;
-        this.geneticAlterationGeneImporter = new GeneticAlterationImporter(geneticProfileId, daoGeneticAlteration);
+        this.updateMode = updateMode;
```

Comment:
```suggestion
        this.isIncrementalUpdateMode = isIncrementalUpdateMode;
```
I keep tripping over this 😅 ...  just `updateMode` does not sound like a boolean to me, but rather like it might contain the mode in which the current data update is running...

Reply:
see here https://github.com/cBioPortal/cbioportal-core/pull/47/commits/79c404129862cf39800091a6046960e9a680f101